Refactor Development section in Contributing guide (#2829)
quickwit-oss/quickwit: CONTRIBUTING.md

Feel free to send your contribution in an unfinished state to get early feedback.
In that case, simply mark the PR with the tag [WIP] (standing for work in progress).

## Signing the CLA
Quickwit is an opensource project licensed a AGPLv3.
It is also distributed under a commercial license by Quickwit, Inc.

Contributors are required to sign a Contributor License Agreement.
The process is simple and fast. Upon your first pull request, you will be prompted to
[sign our CLA by visiting this link](https://cla-assistant.io/quickwit-oss/quickwit).

## PR verification checks

> When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discoverd, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:

Reviewer: ```suggestion
When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discovered, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:
```

> When you submit a pull request to the project, the CI system runs several verification checks. You will be notified by email from the CI system if any issues are discoverd, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:

Reviewer: ```suggestion
When you submit a pull request to the project, the CI system runs several verification checks. After your PR is merged, a more exhaustive list of test will be run.

You will be notified by email from the CI system if any issues are discovered, but if you want to run these checks locally before submitting PR or in order to verify changes you can use the following commands in the root directory:
```

Author: I should have passed it through a spellchecker I guess 😄

> 1. To veryfy that all tests are passing run `make test-all`.

Reviewer: ```suggestion
1. To verify that all tests are passing run `make test-all`.
```

> 2. To fix code style and foramt as well as catch common mistakes run `make fix`. Alternatively, run `make -k test-all docker-compose-down` to tear down the Docker services after running all the tests.

Reviewer: ```suggestion
2. To fix code style and format as well as catch common mistakes run `make fix`. Alternatively, run `make -k test-all docker-compose-down` to tear down the Docker services after running all the tests.
```
